### PR TITLE
Singlepulseextractor output control

### PR DESCRIPTION
**Pull request by dneise**
solves #182 and #183

### CODE REVIEW

**Comment by jebuss** [comment]
singlepeextactor

Here is a screenshot of the current effect off the signal reconstruction. Looks good, if you ask me!

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 1]
```diff
@@ -13,6 +13,23 @@
...
+ * |                                                                       |
+ * |         |................ extraction window .................|        |
+ * |         | <---------- length = 225 ------------------------> |        |
+ * |         |                                                    |        |
+ * |         |       |.. output window ..|                        |        |
+ * 0        20       | <- length=100 ->  |                       245      300
+ *                  30                  130
+ *
  * Created by Sebastian Mueller
  * and some modifications from Jens Buss
```
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: add dneise here

File: src/main/java/fact/features/singlePulse/timeSeriesExtraction/SinglePulseExtractor.java [hunk 2]
```diff
@@ -46,10 +46,28 @@ public int numberOfPulses() {
...
         public double timeSeriesBaseLine() {
-            DescriptiveStatistics statistics = 
+            DescriptiveStatistics statistics =
                 new DescriptiveStatistics(timeSeriesAfterExtraction);
             return statistics.getMean();
-        }    
+        }
+
+        /**
+         * @param: startSlice is with respect to extraction window start.
```
Comment (MaxNoe) [inline src/main/java/fact/features/singlePulse/timeSeriesExtraction/SinglePulseExtractor.java]: This javadocstring is not correct. See http://www.oracle.com/technetwork/articles/java/index-137868.html

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 3]
```diff
@@ -21,8 +38,8 @@
     private String dataKey = null;
 
     @Parameter(
-        required = true, 
-        description =   "output key, a list of lists of arrival slices of "+ 
+        required = true,
+        description =   "output key, a list of lists of arrival slices of "+
```
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: This not a list of lists anymore, its a 2dim in array

**Comment by jebuss** [comment]
Well, at first I thought it is ok to use this multiplication for the sake of reducing the size of a single value instead of all values in the data item. But, with a second thought, I have to agree to what Max said. Its really not that intuitive and maybe we don't need to do this for a single variable, since we probably won't use the single pe extraction together with other feature extractions where we need a higher significance than two digits.

**Comment by dneise** [comment]
Situation pre this PR:

 - keyname: "BaseLine"
 - unit: "Single Photon Amplitude"
 - number of keys with the same unit: 0

Situation after this PR:

 - keyname: "CentiBaseLines"
 - units: "1% of Single Photon Amplitude"
 - number of keys with the same unit: 0

**Comment by MaxNoe** [comment]
Not the point. We always rage about contra-intuitive formats and units. Now we start the same here (without any documentation about the matter).

**Comment by dneise** [comment]
I see your point.

> We could adapt the JSONWriter to allow it to set the significant digits per key not just globally. However, I think we can reduce the number of significant digits quite dramatically in general (4 digits should be enough for everything we write out).

This sounds like the way to go. I looked at the code ... doesn't look too easy ... in fact I would not even know where to start. 

At the moment I run a test with `doubleSignDigits = 3` ... just to see where we end up.

**Comment by dneise** [comment]
I cherry picked the agreed upon parts into another branch and opened a dedicated PR for that.
see #185

**Comment by dneise** [comment]
I added two processors which allow us to change the unit of Baseline outside of the processor creating Baseline in another PR. They are pretty general, so I guess it makes sense to give them their own PR.

see #186 

So this PR can be closed now.

**Pull request closed without merging**